### PR TITLE
feat(anna): blind gossip

### PR DESCRIPTION
**Pull request by rohitkulshreshtha**
This change implements a blind gossip protocol where the gossipers don't receive (or care to receive) responses from the systems they gossip with. 

It also adds a deterministic testing framework for testing gossip.

Once request-reply is added, it can stop being blind.

### CODE REVIEW

File: datastores/gossip_kv/server/simulation.rs [hunk 1]
```diff
@@ -0,0 +1,430 @@
...
+//!  │ │ │  └──────┘  └──────┘  └──────┘ │  │              │ │  └──────┘  └──────┘  └──────┘ │ │ │
+//!  │ │ └───────────────────────────────┘  │              │ └───────────────────────────────┘ │ │
+//!  │ └────────────────────────────────────┼──────────────┼───────────────────────────────────┘ │
+//!  │                                    ┌─┼──────────────┼─┐                                   │
+//!  │                                    │ └--------------┘ │                                   │
+//!  │                                    │ NETWORK MESSAGE  │                                   │
+//!  │                                    │    PROCESSING    │                                   │
+//!  │                                    └──────────────────┘                                   │
+//!  └───────────────────────────────────────────────────────────────────────────────────────────┘
+//! ```
```
Comment (MingweiSamuel) [inline datastores/gossip_kv/server/simulation.rs:46]: niceee

Comment (MingweiSamuel) [reply datastores/gossip_kv/server/simulation.rs:46]: I guess we will want to make this module a feature of hydroflow eventually?

**Review by MingweiSamuel — APPROVED** [review]
Nice!

File: datastores/gossip_kv/server/simulation.rs (hunk 1 above)
Comment (rohitkulshreshtha) [reply datastores/gossip_kv/server/simulation.rs:46]: https://github.com/hydro-project/hydroflow/pull/1416

**Pull request merged**